source/common/stats/utility.cc

```diff
@@ -126,6 +126,19 @@ TextReadout& textReadoutFromStatNames(Scope& scope, const StatNameVec& elements,
   return scope.textReadoutFromStatNameWithTags(StatName(joined.get()), tags);
 }
 
+absl::flat_hash_set<std::string>& customStatNamespaces() {
+  MUTABLE_CONSTRUCT_ON_FIRST_USE(absl::flat_hash_set<std::string>);
```

Comment:
Please don't use a static for this -- it makes tests non-hermetic.

This should be owned by a factory with lifetime corresponding to the server, and pass the StatNamespace object into the sinks.